api/src/test/java/com/bjs/bjsapi/controllers/StudentScoreControllerTest.java

```diff
@@ -146,4 +151,59 @@ public void test_calculation_general() {
 
 		assertThat(integer).isEqualTo(1526);
 	}
+
+	@Test
+	public void test_classification() {
+		Student student = new Student();
+		Date studentsBirthday = new Date(2000, 01, 01);
+		student.setFemale(false);
+		student.setBirthDay(studentsBirthday);
+
+		SportResult resultRUN_100 = new SportResult();
+		resultRUN_100.setDiscipline(DisciplineType.RUN_100);
+		resultRUN_100.setResult(12.00F);
+
+		SportResult resultRUN_800 = new SportResult();
+		resultRUN_800.setDiscipline(DisciplineType.RUN_800);
+		resultRUN_800.setResult(210.00F);
+
+		SportResult resultJUMP = new SportResult();
+		resultJUMP.setDiscipline(DisciplineType.LONG_JUMP);
+		resultJUMP.setResult(4.20F);
+
+		SportResult resultTHROW_200 = new SportResult();
+		resultTHROW_200.setDiscipline(DisciplineType.BALL_THROWING_200);
+		resultTHROW_200.setResult(35.00F);
+
+		doReturn(Arrays.asList(resultJUMP, resultRUN_100, resultRUN_800, resultTHROW_200)).when(sportResultRepository).findByStudent(student);
+
+		doReturn(4.00620).when(calculationInformationService).getAValue(false, DisciplineType.RUN_100);
+		doReturn(2.02320).when(calculationInformationService).getAValue(false, DisciplineType.RUN_800);
+		doReturn(1.09350).when(calculationInformationService).getAValue(false, DisciplineType.LONG_JUMP);
+		doReturn(1.41490).when(calculationInformationService).getAValue(false, DisciplineType.BALL_THROWING_200);
+
+		doReturn(0.00656).when(calculationInformationService).getCValue(false, DisciplineType.RUN_100);
+		doReturn(0.00647).when(calculationInformationService).getCValue(false, DisciplineType.RUN_800);
+		doReturn(0.00208).when(calculationInformationService).getCValue(false, DisciplineType.LONG_JUMP);
+		doReturn(0.01039).when(calculationInformationService).getCValue(false, DisciplineType.BALL_THROWING_200);
+
+		doReturn(1275).when(classificationInformationService).getVictoryValue(false, 19);
+		doReturn(1550).when(classificationInformationService).getHonorValue(false, 19);
```

Comment:
All this code can be saved by using a Mockito `spy`, I can show you this feature some time.

Reply:
That would be cool.